### PR TITLE
Add setting "ip_configuration" to allow private IPs

### LINKED ISSUE #12: Add support for setting ip_configuration on jobs
We should add an `ip_configuration` variable which allows overriding the job IP configuration to allow private IPs.

### PR DESCRIPTION
**Pull request by paulpalamarchuk**
* Fix #12

### CODE REVIEW

**Comment by paulpalamarchuk** [comment]
Also created PR [fix provider's docs](https://github.com/terraform-providers/terraform-provider-google/pull/4806)

File: examples/dlp_api_example/main.tf [hunk 1]
```diff
@@ -15,7 +15,7 @@
  */
 
 provider "google" {
-  version = "~> 2.8.0"
+  version = "~> 2.18"
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply examples/dlp_api_example/main.tf]: I don't see any reason to limit version only to `>= 2.18.0, < 2.19`

File: main.tf [hunk 2]
```diff
@@ -31,6 +31,7 @@ resource "google_dataflow_job" "dataflow_job" {
     "/(.*)/regions/(.*)/",
     "regions/$2",
   )
-  machine_type = var.machine_type
+  machine_type     = var.machine_type
+  ip_configuration = var.ip_configuration == "" ? null : var.ip_configuration
```
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply main.tf]: updated

**Comment by paulpalamarchuk** [comment]
@kopachevsky Not sure if it's needed.

File: examples/dlp_api_example/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (morgante) [reply examples/dlp_api_example/main.tf]: That is our practice, so please follow it.

File: examples/simple_example/main.tf [hunk 3]
```diff
@@ -15,7 +15,7 @@
  */
 
 provider "google" {
-  version = "~> 2.8.0"
+  version = "~> 2.18"
```
Comment (morgante) [inline examples/simple_example/main.tf]:
```suggestion
  version = "~> 2.18.0"
```

File: versions.tf [hunk 4]
```diff
@@ -16,4 +16,8 @@
 
 terraform {
   required_version = ">= 0.12"
+  required_providers {
+    google = "~> 2.18"
```
[the remark this answers is not in the document]
Comment (morgante) [reply versions.tf]: Actually this looser constraint is fine for the `required_providers` setting.

File: examples/dlp_api_example/main.tf (hunk 1 above)
[the remark this answers is not in the document]
Comment (paulpalamarchuk) [reply examples/dlp_api_example/main.tf]: Changed

File: examples/simple_example/main.tf (hunk 3 above)
Comment (paulpalamarchuk) [reply examples/simple_example/main.tf]: Changed

**Pull request merged**